onnxscript/function_libs/tools/torch_lib/deduce_type_constraints.py

```diff
@@ -0,0 +1,403 @@
+from __future__ import annotations
```

Comment:
As discussed it would be helpful to include an explanation of the binding and updating process as documentation.